File: src/builder/java/champion/Champion.java
package builder.java.champion;

/**
 * Created by kvirus on 2019/6/13 18:07
 * Email @ caoyingkui@pku.edu.cn
 * <p>
 * |   *******    **      **     **     **
 * |  **            **  **       **  **
 * |  **              **         ***
 * |  **              **         **  **
 * |   *******        **         **     **
 */

public abstract class Champion {
    public static enum ChampionType {
        TANK,           // 坦克
        FIGHTER,        // 战士
        SLAYER,         // 刺客
        MAGE,           // 法师
        CONTROLLER,     // 辅助
        MARKSMAN        // 射手
    }

    public String nickname;     // 英雄别称
    public String slogon;       // 座右铭
    public String continent;    // 出生地
    public ChampionType type;   // 类型

    public String getInfo() {
        StringBuilder info = new StringBuilder();
        info.append("\tnickname: " + nickname + "\n");
        info.append("\tslogon: " + slogon + "\n");
        info.append("\tcontinent: " + continent + "\n");
        info.append("\ttype: " + type + "\n");
        return info.toString();
    }
}

File: src/builder/kotlin/champion/Champion.kt
package builder.kotlin.champion

/**
 * Created by kvirus on 2019/6/13 19:31
 * Email @ caoyingkui@pku.edu.cn
 *
 * |   *******    **      **     **     **
 * |  **            **  **       **  **
 * |  **              **         ***
 * |  **              **         **  **
 * |   *******        **         **     **
 */

enum class ChampionType {
    TANK,           // 坦克
    FIGHTER,        // 战士
    SLAYER,         // 刺客
    MAGE,           // 法师
    CONTROLLER,     // 辅助
    MARKSMAN,        // 射手
    UNDIFINED
}

open class Champion {
    open var nickname: String = ""
    open var slogon: String = ""
    open var continent: String = ""
    open var type: ChampionType = ChampionType.UNDIFINED

    open fun getInfo(): String {
        return "\tnickname: $nickname\n" +
                "\tslogon: $slogon\n" +
                "\tcontinent: $continent\n" +
                "\ttype: $type";
    }
}